src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/Error/SqlExtendedQueryTagErrorStoreV4.cs

```diff
@@ -0,0 +1,115 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Features.Client;
+using Microsoft.Health.SqlServer.Features.Storage;
+
+namespace Microsoft.Health.Dicom.SqlServer.Features.ExtendedQueryTag.Error
+{
+    internal class SqlExtendedQueryTagErrorStoreV4 : SqlExtendedQueryTagErrorStoreV3
+    {
+
+        public SqlExtendedQueryTagErrorStoreV4(
+           SqlConnectionWrapperFactory sqlConnectionWrapperFactory,
+           ILogger<SqlExtendedQueryTagErrorStoreV4> logger)
+        {
+            EnsureArg.IsNotNull(sqlConnectionWrapperFactory, nameof(sqlConnectionWrapperFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            ConnectionWrapperFactory = sqlConnectionWrapperFactory;
+            Logger = logger;
+        }
+
+        public override SchemaVersion Version => SchemaVersion.V4;
+
+        protected SqlConnectionWrapperFactory ConnectionWrapperFactory { get; }
+
+        protected ILogger Logger { get; }
+
+        public override async Task<int> AddExtendedQueryTagErrorAsync(
+            int tagKey,
+            int errorCode,
+            long watermark,
+            DateTime createdTime,
+            CancellationToken cancellationToken = default)
+        {
+            using SqlConnectionWrapper sqlConnectionWrapper = await ConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken);
+            using SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand();
+            VLatest.AddExtendedQueryTagError.PopulateCommand(
+                sqlCommandWrapper,
+                tagKey,
+                errorCode,
+                watermark,
+                createdTime);
+            try
+            {
+                return (int)await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken);
+            }
+            catch (SqlException e)
+            {
+                switch (e.Number)
+                {
+                    case SqlErrorCodes.Conflict:
+                        throw new ExtendedQueryTagErrorAlreadyExistsException();
```

Comment:
Do we care about this scenario to warrant an error? How could this happen? Multiple activities indexing the same range?